### PR TITLE
Feature to load settings from a configuration file

### PR DESCRIPTION
**Pull request by joBr99**
I started to implement the possibility to load configurations.
I added a setup step where you can choose from found json configuration files.
It is also possible to choose the normal configuration from this menu.

Related to the following issues:
#15 
#11

### CODE REVIEW

**Comment by bnw** [comment]
Hey joBr99, thank you so much! I tested your code and it worked flawlessly. I will just add a few minor comments.

File: app/index.php [hunk 1]
```diff
@@ -19,19 +20,73 @@
...
 
 switch ((string)$current_step) {
-    case Step::STEP1_COLLECTING_DATA:
+    case Step::STEP0_SETUP:
+        $configuration_files = array();
+        $dirs = array('/app/configurations', 'configurations');
+        foreach($dirs as $dir){
+            if (file_exists($dir))
+                $configuration_files = array_merge($configuration_files,
+                    preg_filter('/^/', $dir.'/', array_diff(scandir($dir), array('.', '..') )));
```
Comment (bnw) [inline app/index.php:36]: Does this work on Windows? Maybe use `$dir.DIRECTORY_SEPARATOR` instead.

File: app/public/html/setup.twig [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            <legend>Data Source</legend>
+            <div class="form-group">
+                <label for="bank_sfa">Choose file from configurations folder or </label>
+                <select class="form-control" id="data_collect_mode" name="data_collect_mode" required>
+                    {% for file in files %}
+                        <option value="{{ file }}">{{ file }}</option>
+                    {% endfor %}
+                    <option value="createNewDataset">Configure Dataset</option>
+                </select>
+                <small class="form-text text-muted">Choose an configuration file or configure it manual.</small>
```
Comment (bnw) [inline app/public/html/setup.twig:19]: I believe the wording is a little bit confusing: "Configure Dataset" sounds like one creates a configuration file, even tho no configuration file will be generated.
What do you think about the following:
Label: `Configuration`
createNewDataset-Option: `Manually fill out the configuration form`
Help: `Choose a method to configure the importer. You can either select a configuration file that you created previously, or you can use the configuration form.`

File: app/index.php (hunk 1 above)
Comment (joBr99) [reply app/index.php:36]: Worked for me on windows with an xampp apache webserver and with the docker container, but it should be a bit cleaner with DIRECTORY_SEPARATOR, so i've no issue with that.

File: app/public/html/setup.twig (hunk 2 above)
Comment (joBr99) [reply app/public/html/setup.twig:19]: Yeah, change the wording as you like, if we implement a option to save new configuration files later, we can still change it.

**Comment by psistorm** [comment]
If I leave the user id and pin empty it fails with an Exception:
`Fatal error: Uncaught InvalidArgumentException: pin cannot be empty in /vendor/nemiah/php-fints/lib/Fhp/Options/Credentials.php:35 Stack trace: #0 /app/FinTsFactory.php(26): Fhp\Options\Credentials::create('bohmer.push', '') #1 /app/index.php(67): App\FinTsFactory::create_from_session(Object(Symfony\Component\HttpFoundation\Session\Session)) #2 {main} thrown in /vendor/nemiah/php-fints/lib/Fhp/Options/Credentials.php on line 35`

But saving these two could be a security issue in my opinion. It would be good if one can create general bank settings but still allowing to enter user id and pin afterwards. Actually the CSV importer handles it the same. It uses the same form for entering all data like if no preset is created, just already filling out known fields but leaving unknown fields empty.

**Pull request merged**